ding/policy/decision_transformer.py

```diff
@@ -0,0 +1,441 @@
+                )
+
+                # discrete action # TODO
+                # actions = torch.randint(0,self.act_dim,[eval_batch_size, self.max_eval_ep_len, 1], dtype=torch.long, device=self.device)
+
+                states = torch.zeros(
+                    (eval_batch_size, self.max_eval_ep_len, self.state_dim), dtype=torch.float32, device=self.device
+                )
+                rewards_to_go = torch.zeros(
+                    (eval_batch_size, self.max_eval_ep_len, 1), dtype=torch.float32, device=self.device
+                )
+
+                # init episode
+                running_state = self._env.reset()
+                running_reward = 0
+                running_rtg = self.rtg_target / self.rtg_scale
+
+                for t in range(self.max_eval_ep_len):
+
+                    total_timesteps += 1
+
+                    # add state in placeholder and normalize
+                    states[0, t] = torch.from_numpy(running_state).to(self.device)
+                    # states[0, t] = (states[0, t].cpu() - self.state_mean.cpu().numpy()) / self.state_std.cpu().numpy()
+                    states[0, t] = (states[0, t] - self.state_mean) / self.state_std
+
+                    # calcualate running rtg and add it in placeholder
+                    running_rtg = running_rtg - (running_reward / self.rtg_scale)
+                    rewards_to_go[0, t] = running_rtg
+
+                    if t < self.context_len:
+                        _, act_preds, _ = self._learn_model.forward(
+                            timesteps[:, :self.context_len], states[:, :self.context_len],
+                            actions[:, :self.context_len], rewards_to_go[:, :self.context_len]
+                        )
+                        act = act_preds[0, t].detach()
+                    else:
+                        _, act_preds, _ = self._learn_model.forward(
+                            timesteps[:, t - self.context_len + 1:t + 1], states[:, t - self.context_len + 1:t + 1],
+                            actions[:, t - self.context_len + 1:t + 1], rewards_to_go[:, t - self.context_len + 1:t + 1]
+                        )
+                        act = act_preds[0, -1].detach()
+
+                    # if discrete
+                    if not self._cfg.model.continuous:
+                        act = torch.argmax(act)
+                    running_state, running_reward, done, _ = self._env.step(act.cpu().numpy())
+
+                    # add action in placeholder
+                    actions[0, t] = act
+
+                    total_reward += running_reward
+
+                    if render:
+                        self._env.render()
+                    if done:
+                        break
+
+        results['eval/avg_reward'] = total_reward / self.num_eval_ep
+        results['eval/avg_ep_len'] = total_timesteps / self.num_eval_ep
+
+        return results
+
+    def evaluate(self, state_mean=None, state_std=None, render=False):
```

Comment:
abstract this method to a general function in `ding/worker/collect`

Reply:
Actually, this evaluate process is highly coupled with decision transformer, few of codes can be reused by other algos